calculator_test.go

```diff
@@ -7,162 +7,196 @@ import (
+		a, b        float64
+		want        float64
+		test     string
+		errExpected bool
+	}{
+		{a: 2, b: 1, want: 2, errExpected: false, test: "two divided by 1"},
+		{a: 10, b: 5, want: 2, errExpected: false, test: "ten divided by 5"},
+		{a: 10, b: 0, want: 0, errExpected: true, test: "error out on divide by zero"},
+		{a: 10, b: 0, want: 1, errExpected: true, test: "test error case by failing want vs get"},
+		{a: 10, b: 0, want: 1_000_000, errExpected: true, test: "dodgy want to test err"},
+		{a: 10, b: 0, want: 999, errExpected: true, test: "dodgy want to test err"},
+	}
 	t.Parallel()
-	for _, tc := range testCasesDivide {
-		want := tc.want
+	for _, tc := range testCases {
 		got, err := calculator.Divide(tc.a, tc.b)
 		errorReceived := err != nil
 		if errorReceived != tc.errExpected {
-			t.Fatalf("test %s: expected err %t, got %t", tc.comment, tc.errExpected, errorReceived)
+			t.Fatalf("test %s: expected err %t, got %t", tc.test, tc.errExpected, errorReceived)
 		}
-		if want != got {
-			t.Errorf("test %s: want %f, got %f", tc.comment, want, got)
+		if !errorReceived && tc.want != got {
+			t.Errorf("test %s: want %f, got %f", tc.test, tc.want, got)
 		}
 	}
 }
 
 func TestDivideRandom(t *testing.T) {
 	t.Parallel()
 	for i := 0; i < 100; i++ {
-		a := rand.Float64()
-		b := rand.Float64()
+		a := rand.Float64() + 1
+		b := rand.Float64() + 1
 		want := a / b
-		got, _ := calculator.Divide(a, b)
+		got, err := calculator.Divide(a, b)
+		if err != nil {
+			t.Errorf("Divide(%f, %f): want %f, got %f", a, b, want, got)
+		}
 		if want != got {
-			t.Errorf("random test: want %f, got %f", want, got)
+			t.Errorf("Divide(%f, %f): want %f, got %f", a, b, want, got)
 		}
 	}
 }
 
 func TestSqrt(t *testing.T) {
+	testCases := []struct {
+		a, b        float64
+		want        float64
+		test     string
+		errExpected bool
+	}{
+		{a: 16, want: 4, errExpected: false, test: "square root of 16"},
+		{a: -1, want: 0, errExpected: true, test: "error"},
+		{a: -1, want: 0, errExpected: true, test: "error out on negative input"},
+		{a: -1, want: 1, errExpected: true, test: "test error case by failing want vs get"},
+		{a: math.Inf(0), want: math.Inf(0), errExpected: false, test: "test +Inf"},
+	}
 	t.Parallel()
-	for _, tc := range testCasesSqrt {
-		want := tc.want
+	for _, tc := range testCases {
 		got, err := calculator.Sqrt(tc.a)
 		errorReceived := err != nil
 		if errorReceived != tc.errExpected {
-			t.Fatalf("test %s: expected err %t, got %t", tc.comment, tc.errExpected, errorReceived)
+			t.Fatalf("test %s: expected err %t, got %t", tc.test, tc.errExpected, errorReceived)
 		}
-		if want != got {
-			t.Errorf("test %s: want %f, got %f", tc.comment, want, got)
+		if !errorReceived && tc.want != got {
+			t.Errorf("test %s: want %f, got %f", tc.test, tc.want, got)
 		}
 	}
 }
 
 func TestSqrtRandom(t *testing.T) {
 	t.Parallel()
 	for i := 0; i < 100; i++ {
-		a := rand.Float64()
+		a := rand.Float64() + 1
 		want := math.Sqrt(a)
```

Comment:
Of course it's fine here, but in general, if we're _computing_ test expectations, rather than hard-wiring them, we need to be careful we don't just end up testing the same code path against itself! We can assume `math.Sqrt` works, but ideally, we would implement our own Sqrt routine (Newton-Raphson, perhaps) to check it against.